File: docs/snippets/all/archetypes/view_coordinates_simple.cpp
// Set the default orientation for a 3D view.

#include <rerun.hpp>

int main(int argc, char* argv[]) {
    const auto rec = rerun::RecordingStream("rerun_example_view_coordinates");
    rec.spawn().exit_on_failure();

    // Set the 3D view's up direction:
    rec.log_static("world", rerun::ViewCoordinates::RIGHT_HAND_Z_UP);
    rec.log(
        "world/xyz",
        rerun::Arrows3D::from_vectors(
            {{1.0, 0.0, 0.0}, {0.0, 1.0, 0.0}, {0.0, 0.0, 1.0}}
        )
            .with_colors({{255, 0, 0}, {0, 255, 0}, {0, 0, 255}})
    );
}

File: docs/snippets/all/archetypes/view_coordinates_simple.py
"""Set the default orientation for a 3D view."""

import rerun as rr

rr.init("rerun_example_view_coordinates", spawn=True)

rr.log(
    "world", rr.ViewCoordinates.RIGHT_HAND_Z_UP, static=True
)  # Set the 3D view's up direction
rr.log(
    "world/xyz",
    rr.Arrows3D(
        vectors=[[1, 0, 0], [0, 1, 0], [0, 0, 1]],
        colors=[[255, 0, 0], [0, 255, 0], [0, 0, 255]],
    ),
)
